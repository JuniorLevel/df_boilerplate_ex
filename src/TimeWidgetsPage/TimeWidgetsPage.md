```jsx
import React from 'react';
import { Layout } from '../../../lib/components/Layout/Layout';
import TimeWidgetsPanel from '../../../lib/components/TimeWidgetsPanel/TimeWidgetsPanel';
import { buttons } from '../../../lib/components/Container/buttons';

<Layout buttons={buttons} title="Caption" avatar={{ title: 'AV' }}>
	<TimeWidgetsPanel />
</Layout>;
```
