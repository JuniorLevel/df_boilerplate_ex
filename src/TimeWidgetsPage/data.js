export const sidebarButtons = [
	{
		id: 1,
		title: 'Orders',
		onClick: () => {},
	},
	{
		id: 2,
		title: 'Compare',
		onClick: () => {},
	},
	{
		id: 3,
		title: 'Admin',
		onClick: () => {},
	},
];
