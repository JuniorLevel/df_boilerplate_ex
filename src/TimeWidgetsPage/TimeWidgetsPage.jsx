// @flow

import React from 'react';
import { TimeWidgetsPanel, Layout } from '@juniorlevel/df_boilerplate-library';
import { sidebarButtons } from './data';

export const TimeWidgetsPage = (): React$MixedElement => (
	<Layout buttons={sidebarButtons} title="Caption" avatar={{ title: 'AV' }}>
		<TimeWidgetsPanel />
	</Layout>
);
