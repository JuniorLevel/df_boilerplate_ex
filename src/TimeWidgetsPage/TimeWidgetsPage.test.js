import React from 'react';
import { Context as ResponsiveContext } from 'react-responsive';
import { render } from '@testing-library/react';
import { TimeWidgetsPage } from './TimeWidgetsPage';
import '../../__mocks__/fileMock';

describe('TimeWidgetsPage', () => {
	test('should render TimeWidgetsPage component without crashing', () => {
		const { container } = render(<TimeWidgetsPage />);
		expect(container).toBeInTheDocument();
	});
	test('matches the snapshot desktop', () => {
		const { container: desktop } = render(
			<ResponsiveContext.Provider value={{ width: 1920 }}>
				<TimeWidgetsPage />
			</ResponsiveContext.Provider>
		);
		expect(desktop).toMatchSnapshot();
	});
	test('matches the snapshot laptop', () => {
		const { container: laptop } = render(
			<ResponsiveContext.Provider value={{ width: 1024 }}>
				<TimeWidgetsPage />
			</ResponsiveContext.Provider>
		);
		expect(laptop).toMatchSnapshot();
	});
	test('matches the snapshot mobile', () => {
		const { container: mobile } = render(
			<ResponsiveContext.Provider value={{ width: 360 }}>
				<TimeWidgetsPage />
			</ResponsiveContext.Provider>
		);
		expect(mobile).toMatchSnapshot();
	});
});
