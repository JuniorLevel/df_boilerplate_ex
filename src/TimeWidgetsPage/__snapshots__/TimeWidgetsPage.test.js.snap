// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimeWidgetsPage matches the snapshot desktop 1`] = `
<div>
  <div
    class="ant-layout Layoutstyles__StyledLayout-sc-httflb-0 dCkqfg css-1gwfwyx"
  >
    <div
      class="ant-layout css-1gwfwyx"
    >
      <main
        class="ant-layout-content Layoutstyles__StyledContent-sc-httflb-1 eUqUAz css-1gwfwyx"
      >
        <div
          class="TimeWidgetsPanel__StyledDiv-sc-1iizsjd-0 eYgPcl"
        >
          <div
            style="padding: 10px; border: 1px solid black;"
          >
            <div
              class="ant-flex css-1gwfwyx ant-flex-wrap-wrap ant-flex-align-center ant-flex-justify-center"
              style="gap: 10px;"
            >
              <div
                class="ant-flex css-1gwfwyx"
                style="gap: 10px;"
              >
                <span
                  class="ant-typography css-1gwfwyx"
                  style="border: 1px solid black; padding: 10px;"
                >
                  13
                </span>
                <span
                  class="ant-typography css-1gwfwyx"
                  style="border: 1px solid black; padding: 10px;"
                >
                  08
                </span>
                <span
                  class="ant-typography css-1gwfwyx"
                  style="border: 1px solid black; padding: 10px;"
                >
                  99
                </span>
              </div>
              <div
                class="ant-flex css-1gwfwyx"
                style="gap: 10px;"
              >
                <span
                  class="ant-typography css-1gwfwyx"
                  style="border: 1px solid black; padding: 10px;"
                >
                  20
                </span>
                <span
                  class="ant-typography css-1gwfwyx"
                  style="border: 1px solid black; padding: 10px;"
                >
                  55
                </span>
                <span
                  class="ant-typography css-1gwfwyx"
                  style="border: 1px solid black; padding: 10px;"
                >
                  55
                </span>
              </div>
            </div>
          </div>
          <div
            style="border: 1px solid black; padding: 10px; background-color: grey;"
          >
            <div
              class="ant-typography TextFieldWidget__StyledParagraph-sc-v2harq-0 hQDWIX css-1gwfwyx"
            >
              Lorem asdasdas asdasdasd asdasdasd asdasdasdas sadasdasdasd asfdsdfsdfs sdfsdfsd fsdfsdahfd adfhfasdhgasdfg asfdhadffsdafs afdshadfgsadf
            </div>
          </div>
          <div
            class="ant-flex css-1gwfwyx ant-flex-wrap-wrap ant-flex-align-center"
            style="border: 1px solid black; padding: 5px; gap: 5px;"
          >
            <div
              class="ant-flex css-1gwfwyx"
              style="border: 1px solid black; position: relative; padding: 20px;"
            >
              <span
                class="ant-typography css-1gwfwyx"
              >
                Assigned
              </span>
              <span
                class="ant-typography css-1gwfwyx"
                style="position: absolute; top: 0px; left: 5px; font-size: 10px;"
              >
                role
              </span>
            </div>
            <div
              class="ant-flex css-1gwfwyx ant-flex-align-center"
              style="border: 1px solid black; padding: 15px; flex: 1 1 320px; gap: 10px;"
            >
              <span
                class="ant-avatar ant-avatar-circle css-1gwfwyx"
                style="border: 1px solid black; border-radius: 50%; padding: 5px; font-size: 10px; color: black; background-color: white;"
              >
                <span
                  class="ant-avatar-string"
                  style="transform: scale(1);"
                >
                  FIO
                </span>
              </span>
              <span
                class="ant-typography css-1gwfwyx"
              >
                FIO FIO FIO FIO
              </span>
            </div>
          </div>
        </div>
      </main>
    </div>
  </div>
</div>
`;

exports[`TimeWidgetsPage matches the snapshot laptop 1`] = `
<div>
  <div
    class="ant-layout Layoutstyles__StyledLayout-sc-httflb-0 dCkqfg css-1gwfwyx"
  >
    <div
      class="ant-layout css-1gwfwyx"
    >
      <main
        class="ant-layout-content Layoutstyles__StyledContent-sc-httflb-1 eUqUAz css-1gwfwyx"
      >
        <div
          class="TimeWidgetsPanel__StyledDiv-sc-1iizsjd-0 eYgPcl"
        >
          <div
            style="padding: 10px; border: 1px solid black;"
          >
            <div
              class="ant-flex css-1gwfwyx ant-flex-wrap-wrap ant-flex-align-center ant-flex-justify-center"
              style="gap: 10px;"
            >
              <div
                class="ant-flex css-1gwfwyx"
                style="gap: 10px;"
              >
                <span
                  class="ant-typography css-1gwfwyx"
                  style="border: 1px solid black; padding: 10px;"
                >
                  13
                </span>
                <span
                  class="ant-typography css-1gwfwyx"
                  style="border: 1px solid black; padding: 10px;"
                >
                  08
                </span>
                <span
                  class="ant-typography css-1gwfwyx"
                  style="border: 1px solid black; padding: 10px;"
                >
                  99
                </span>
              </div>
              <div
                class="ant-flex css-1gwfwyx"
                style="gap: 10px;"
              >
                <span
                  class="ant-typography css-1gwfwyx"
                  style="border: 1px solid black; padding: 10px;"
                >
                  20
                </span>
                <span
                  class="ant-typography css-1gwfwyx"
                  style="border: 1px solid black; padding: 10px;"
                >
                  55
                </span>
                <span
                  class="ant-typography css-1gwfwyx"
                  style="border: 1px solid black; padding: 10px;"
                >
                  55
                </span>
              </div>
            </div>
          </div>
          <div
            style="border: 1px solid black; padding: 10px; background-color: grey;"
          >
            <div
              class="ant-typography TextFieldWidget__StyledParagraph-sc-v2harq-0 hQDWIX css-1gwfwyx"
            >
              Lorem asdasdas asdasdasd asdasdasd asdasdasdas sadasdasdasd asfdsdfsdfs sdfsdfsd fsdfsdahfd adfhfasdhgasdfg asfdhadffsdafs afdshadfgsadf
            </div>
          </div>
          <div
            class="ant-flex css-1gwfwyx ant-flex-wrap-wrap ant-flex-align-center"
            style="border: 1px solid black; padding: 5px; gap: 5px;"
          >
            <div
              class="ant-flex css-1gwfwyx"
              style="border: 1px solid black; position: relative; padding: 20px;"
            >
              <span
                class="ant-typography css-1gwfwyx"
              >
                Assigned
              </span>
              <span
                class="ant-typography css-1gwfwyx"
                style="position: absolute; top: 0px; left: 5px; font-size: 10px;"
              >
                role
              </span>
            </div>
            <div
              class="ant-flex css-1gwfwyx ant-flex-align-center"
              style="border: 1px solid black; padding: 15px; flex: 1 1 320px; gap: 10px;"
            >
              <span
                class="ant-avatar ant-avatar-circle css-1gwfwyx"
                style="border: 1px solid black; border-radius: 50%; padding: 5px; font-size: 10px; color: black; background-color: white;"
              >
                <span
                  class="ant-avatar-string"
                  style="transform: scale(1);"
                >
                  FIO
                </span>
              </span>
              <span
                class="ant-typography css-1gwfwyx"
              >
                FIO FIO FIO FIO
              </span>
            </div>
          </div>
        </div>
      </main>
    </div>
  </div>
</div>
`;

exports[`TimeWidgetsPage matches the snapshot mobile 1`] = `
<div>
  <div
    class="ant-layout Layoutstyles__StyledLayout-sc-httflb-0 dCkqfg css-1gwfwyx"
  >
    <div
      class="ant-layout css-1gwfwyx"
    >
      <main
        class="ant-layout-content Layoutstyles__StyledContent-sc-httflb-1 eUqUAz css-1gwfwyx"
      >
        <div
          class="TimeWidgetsPanel__StyledDiv-sc-1iizsjd-0 eYgPcl"
        >
          <div
            style="padding: 10px; border: 1px solid black;"
          >
            <div
              class="ant-flex css-1gwfwyx ant-flex-wrap-wrap ant-flex-align-center ant-flex-justify-center"
              style="gap: 10px;"
            >
              <div
                class="ant-flex css-1gwfwyx"
                style="gap: 10px;"
              >
                <span
                  class="ant-typography css-1gwfwyx"
                  style="border: 1px solid black; padding: 10px;"
                >
                  13
                </span>
                <span
                  class="ant-typography css-1gwfwyx"
                  style="border: 1px solid black; padding: 10px;"
                >
                  08
                </span>
                <span
                  class="ant-typography css-1gwfwyx"
                  style="border: 1px solid black; padding: 10px;"
                >
                  99
                </span>
              </div>
              <div
                class="ant-flex css-1gwfwyx"
                style="gap: 10px;"
              >
                <span
                  class="ant-typography css-1gwfwyx"
                  style="border: 1px solid black; padding: 10px;"
                >
                  20
                </span>
                <span
                  class="ant-typography css-1gwfwyx"
                  style="border: 1px solid black; padding: 10px;"
                >
                  55
                </span>
                <span
                  class="ant-typography css-1gwfwyx"
                  style="border: 1px solid black; padding: 10px;"
                >
                  55
                </span>
              </div>
            </div>
          </div>
          <div
            style="border: 1px solid black; padding: 10px; background-color: grey;"
          >
            <div
              class="ant-typography TextFieldWidget__StyledParagraph-sc-v2harq-0 hQDWIX css-1gwfwyx"
            >
              Lorem asdasdas asdasdasd asdasdasd asdasdasdas sadasdasdasd asfdsdfsdfs sdfsdfsd fsdfsdahfd adfhfasdhgasdfg asfdhadffsdafs afdshadfgsadf
            </div>
          </div>
          <div
            class="ant-flex css-1gwfwyx ant-flex-wrap-wrap ant-flex-align-center"
            style="border: 1px solid black; padding: 5px; gap: 5px;"
          >
            <div
              class="ant-flex css-1gwfwyx"
              style="border: 1px solid black; position: relative; padding: 20px;"
            >
              <span
                class="ant-typography css-1gwfwyx"
              >
                Assigned
              </span>
              <span
                class="ant-typography css-1gwfwyx"
                style="position: absolute; top: 0px; left: 5px; font-size: 10px;"
              >
                role
              </span>
            </div>
            <div
              class="ant-flex css-1gwfwyx ant-flex-align-center"
              style="border: 1px solid black; padding: 15px; flex: 1 1 320px; gap: 10px;"
            >
              <span
                class="ant-avatar ant-avatar-circle css-1gwfwyx"
                style="border: 1px solid black; border-radius: 50%; padding: 5px; font-size: 10px; color: black; background-color: white;"
              >
                <span
                  class="ant-avatar-string"
                  style="transform: scale(1);"
                >
                  FIO
                </span>
              </span>
              <span
                class="ant-typography css-1gwfwyx"
              >
                FIO FIO FIO FIO
              </span>
            </div>
          </div>
        </div>
      </main>
    </div>
  </div>
</div>
`;
